app/code/Magento/Translation/etc/di.xml

```diff
@@ -44,6 +44,7 @@
             <argument name="renderers" xsi:type="array">
                 <item name="translation" xsi:type="object">Magento\Framework\Phrase\Renderer\Translate</item>
                 <item name="placeholder" xsi:type="object">Magento\Framework\Phrase\Renderer\Placeholder</item>
+                <item name="messageFormatter" xsi:type="object">Magento\Framework\Phrase\Renderer\MessageFormatter</item>
```

Comment:
As ICU message format also allows to handle variables (aka placeholders) maybe it would be better to place it before Placeholder renderer. With this approach, we will be able to deprecate Placeholder renderer and after some period of time, we should come to the point where all variables substitution happens at MessageFormatter renderer and not use Placeholder. Then we will be able to remove Placeholder render and declare that ICU format is the only one to use in phrases.
@navarr  do you see  any drawbacks of having MessageFormatter before Placeholder?

Reply:
I can't think of any drawbacks.  I think in general it will work the same either way - MessageFormatter variables are a different format.